### ISSUE TITLE
How to specify cell instances?

### ISSUE THREAD

**Issue by frothga** [NeuroML/NeuroML2#109]
Looking through neuroml_dev/NeuroML2/examples, there are several different ways that preCell/postCell/target attributes get filled out:

connection -> xpath -- with comment that prefer number
connectionWD -> xpath[number]
electricalConnection -> number
electricalConnectionInstance -> xpath
continuousConnection -> number
continuousConnectionInstance -> xpath
input -> xpath -- with comment that prefer number
explicitInput -> name[number]

In summary, it seems that "Instance" type connections get an xpath, while everything else would ideally get a simple number, except that some limitation prevents that from being processed properly. Also, there are two forms of xpath, and one example that appears like an xpath but doesn't reference its container.

Is there a definitive rule on what kind of ID to use where?

When will it be safe to use simple integer IDs?

**Comment by pgleeson** [comment]
@frothga yes, this is far from ideal, and really down to the types of paths jLEMS enforced, so that it could interpret the NeuroML as valid LEMS (and so simulate the network connections based on Componenttype definitions).

The issue is summarised here: https://github.com/NeuroML/NeuroML2/issues/90.

This will all be cleaned up eventually, but not in the next release, so unfortunately it's best just to test with the various examples available. 

If you do work in Java, there is a helper methods here which can be used to interpret the paths: https://github.com/NeuroML/org.neuroml.export/blob/development/src/main/java/org/neuroml/export/utils/Utils.java#L119. Note also this helper class for interpreting paths in LEMS files to variables: https://github.com/NeuroML/org.neuroml.export/blob/development/src/main/java/org/neuroml/export/utils/LEMSQuantityPath.java#L308
